### ISSUE TITLE
Factoriolab is not accessible with the NVDA screen reader

### ISSUE THREAD

**Issue by jareds** [factoriolab/factoriolab#1803]
**Describe the bug**
Factoriolab has many accessibility issues when using the NVDA screen reader on WIndows with the Brave or Chrome browser. A few are
* There are unlabeled buttons under the settings heading.
* There are unlabeled check boxes.
* When searching for something to add as a requirement the buttons are unlabeled so I can't tell what I'm adding.
**To Reproduce**

1. Insure you are using the NVDA screen reader.
2. Go to https://factoriolab.github.io
3. Attempt to use the application.
**Expected behavior**
I expect the app to ahve everything labeled correctly, and keyboard navigation to work as expected with out duplicate items.

**Screenshots**
n/a

**Desktop (please complete the following information):**

- OS: Windows 11
- Browser Chrome, Brave, Firefox
- Version Occurs on all versions of all mentioned browsers over the passed several months.

**Additional context**
I play Factorio using the (Factorio Access mod)[https://github.com/Factorio-Access/FactorioAccess] I realize that this issue is not a simple fix, and don't expect it to be quickly completed. I am willing to open pr's to fix issues I find while using the app. I created this issue to discuss if accessibility specific pr's are desired before I start sending pr's with no context.

**Comment by dcbroad3** [comment]
Well this is kind of exciting. The current main site wasn't really built for accessibility, and I was never sure if anyone with those needs would be playing those games and using the calculator. I do have experience building accessible UI, though, and it's something I've been wanting to improve. 
There is a new version of the calculator built more with accessibility in mind, although it's unfinished and I haven't really done a pass using NVDA to make sure things look right. It can be found at https://factoriolab.github.io/next. If there are things to improve, I would like to focus on that version of the site, since I hope to publish it and replace the existing UI.
You're also welcome to contribute, but I would recommend contributing to the next branch as well.
It's been a work in progress for many months, but I do hope to find time to wrap it up soon.
